main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -34,13 +34,14 @@ object BuildServerProtocol {
   )
 
   lazy val globalSettings: Seq[Def.Setting[_]] = Seq(
-    bspWorkspace := Def.taskDyn {
-      val structure = Keys.buildStructure.value
-      val scopes: Seq[Scope] = structure.allProjectRefs.flatMap { ref =>
-        bspTargetConfigs.toSeq.map(name => Scope.Global.in(ref, ConfigKey(name)))
+    bspWorkspace := Def.settingDyn {
+      val loadedBuild = Keys.loadedBuild.value
+      val scopes: Seq[Scope] = loadedBuild.allProjectRefs.flatMap {
+        case (ref, _) =>
+          bspTargetConfigs.toSeq.map(name => Scope.Global.in(ref, ConfigKey(name)))
       }
-      Def.task {
-        val targetIds = scopes.map(_ / Keys.buildTargetIdentifier).join.value
+      Def.setting {
+        val targetIds = scopes.map(_ / Keys.bspTargetIdentifier).join.value
```

Comment:
See the test failure in https://travis-ci.org/github/sbt/sbt/jobs/689252591#L1013-L1030 with https://github.com/sbt/sbt/blob/develop/sbt/src/sbt-test/project/default-auto-plugins/build.sbt. Some subprojects may not have these keys defined. I wonder if you can do `.?`

Reply:
I've moved the `BSP.globalSettings` from `globalSbtCore` to `globalJvmCore`. This should make the test pass. `BSP.configSettings` and `BSP.handler` are already defined in the JVM plugin and so it is more consistent now.

Reply:
It was not enough because of all global settings are loaded form `BuildDef` ([see comment in the code](https://github.com/adpi2/sbt/blob/b8ba4ae81f46408a8eeeee61727797727e34a8a0/main/src/main/scala/sbt/internal/BuildDef.scala#L22)), so I did what you suggested.